### ISSUE TITLE
[java] AvoidInstantiatingObjectsInLoops: False negative when `new` expression is inside for-loop with array assignment

### ISSUE THREAD

**Issue by mmnhgo** [pmd/pmd#6560]
Affects PMD Version: 7.22.0
Rule: [AvoidInstantiatingObjectsInLoops](https://pmd.github.io/pmd/pmd_rules_java_performance.html#avoidinstantiatingobjectsinloops)

Description:
The **AvoidInstantiatingObjectsInLoops** rule fails to trigger when a `new` expression appears inside a for-loop using array assignment (e.g. `streams[i] = new ...`).

This is a clear false negative — the exact same anti-pattern that is correctly reported in a simple variable assignment is completely missed when the left-hand side is an array element.

Code Sample demonstrating the issue:

```java
import java.io.FileInputStream;
import java.io.IOException;
import java.io.FileNotFoundException;


public class a {

    void TP(String[] files) throws IOException, FileNotFoundException {
        FileInputStream stream;
        for (int i = 0; i < files.length; i++) {
            String file = files[i];
            stream = new FileInputStream(file);      // TP: correctly reported
            stream.close();
        }
    }
    void FN(String[] files) throws IOException, FileNotFoundException {
        FileInputStream[] streams = new FileInputStream[files.length];
        for (int i = 0; i < files.length; i++) {
            String file = files[i];
            streams[i] = new FileInputStream(file);  // FN: should report but missing
            streams[i].close();
        }
    }
}
```
**Expected outcome**:
PMD should report AvoidInstantiatingObjectsInLoops on both lines:

Line 12 (TP method) — correctly reported ✅
Line 20 (FN method) — currently not reported ❌

This is a clear false negative.

Running PMD through: CLI

**Comment by zbynek** [comment]
That's by design. If you want to create an array od objects, using a loop is a perfectly OK way to do it (see e.g. https://github.com/pmd/pmd/pull/3394 ).

There are other issues with the code sample that could be flagged (the array elements are assigned, but effectively never querried), but those should go into a separate issue.
